packages/manager-react-components/src/hooks/date/useFormattedDate.ts

```diff
@@ -0,0 +1,46 @@
+import { useTranslation } from 'react-i18next';
+
+export const defaultUnknownDateLabel = 'N/A';
+
+export enum DateFormat {
+  /**
+   * dd/MM/YYYY
+   */
+  compact = 'compact',
+  /**
+   * dd abbreviated month YYYY
+   */
+  display = 'display',
+  /**
+   * dd month YYYY
+   */
+  fullDisplay = 'fullDisplay',
+}
+
+export const useFormattedDate = ({
+  dateString,
+  defaultLocale = 'FR-fr',
+  unknownDateLabel = defaultUnknownDateLabel,
+  format = DateFormat.display,
+}: {
+  dateString: string;
+  unknownDateLabel?: string;
+  defaultLocale?: string;
+  format?: DateFormat;
+}) => {
+  const { i18n } = useTranslation('date-utils');
+  const date = new Date(dateString);
+  const locale = i18n?.language?.replace('_', '-') || defaultLocale;
+
+  if (date.toString() === 'Invalid Date') {
+    return unknownDateLabel;
```

Comment:
As `unknownDateLabel` is optionnal maybe add : 

I don't known if it's relevant

```suggestion
    return unknownDateLabel ?? DEFAULT_INVALID_DATE_LABEL;
```

Reply:
Ah yes ! My bad